modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -174,6 +184,9 @@ public void run(Context context) {
 
         try {
             exec();
+            if (context.isDebugEnabled()) {
+                VMDebugManager.getInstance().setDone(true);
```

Comment:
Before setting done true. need to check `context.actionInfo != null`, because `exec()` can return due to a non-blocking action invocation.